ui/src/app/services/sign-in.service.ts

```diff
@@ -38,6 +40,7 @@ export class SignInService {
     this.auth2 = this.makeAuth2();
     this.user = this.makeUserSubject();
     this.user.subscribe(newUserDetails => {
+      window.sessionStorage.setItem(SIGNED_IN_USER, JSON.stringify(newUserDetails));
```

Comment:
Session Storage has some nice properties: it's isolated to this tab and does not outlive a browser session. (So this code still flashes if you copy the URL and paste into a new tab or window — this, too , could be mitigated by changing `sessionStorage` --> `localStorage`)